.github/workflows/test.yml

```
@@ -0,0 +1,85 @@
+name: Test
+on:
+  push:
+  pull_request:
+    types: [reopened]
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/setup-docker-compose
+        with:
+          # Only this step sets the cache for future runs
+          cache-to: type=gha,mode=max
+      - name: Lint
+        run: docker-compose run --no-deps web yarn lint:nofix
+
+  frontend:
+    name: Frontend
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/setup-docker-compose
+      - name: Test frontend
+        run: docker-compose run --no-deps web yarn test:js:ci
+      - name: Upload coverage results
+        uses: actions/upload-artifact@v3
+        with:
+          name: frontend-coverage
+          path: |
+            coverage
+
+  backend:
+    name: Backend
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: setup
+        uses: ./.github/setup-docker-compose
+      - name:
+          Copy dist folder (contains the index.html file for Django templates)
+        run:
+          docker cp $(docker create ${{ steps.setup.outputs.tag }}):/app/dist .
```

I'd think in this project this should be `/app/dist/prod` given the settings in `Dockerfile` and `webpack.prod.js` -- but I guess it works this way? 😕 

```suggestion
          docker cp $(docker create ${{ steps.setup.outputs.tag }}):/app/dist/prod .
```

---

Fixed in b5467deddab77a622a98c8033ef2d27decf87e2e

---

@jerivas Can you clarify why this step is needed? I'm confused how this was working both before and after this most recent change. It seems like the `yarn test:py` command wouldn't know where to find this file in any case.

---

Honestly this happened in oddbooks, so it might not be necessary here. The problem was that when the image was built the `dist/` folder only existed inside the container, not the host filesystem. When `docker-compose run` is executed the host filesystem is mounted and overrides the entire `/app` folder in the container, and we lose the `dist/` folder. Copying out to the host filesystem ensures we always have it available.

---

It was working before because `dist` only contains `prod` anyways, so copying either achieves the same effect

---

Ah, I see. So running the python tests must require that the `index.html` file exists? I can't seem to reproduce that locally... Or maybe we just need the _existence_ of `dist/prod`? If that's the case, maybe `mkdir -p dist/prod` would be clearer?